core/services/workflows/syncer/handler.go

```diff
@@ -130,52 +138,53 @@ func (h *eventHandler) Handle(ctx context.Context, event WorkflowRegistryEvent)
 	case WorkflowActivatedEvent:
 		return h.workflowActivatedEvent(ctx, event)
 	default:
-		return fmt.Errorf("event type unsupported: %v", event.EventType)
+		return fmt.Errorf("event type unsupported: %v", event.GetEventType())
 	}
 }
 
 // workflowRegisteredEvent handles the WorkflowRegisteredEvent event type.
 // TODO: Implement this method
 func (h *eventHandler) workflowRegisteredEvent(
 	_ context.Context,
-	_ WorkflowRegistryEvent,
+	_ event,
 ) error {
-	return ErrNotImplemented
+	h.stats.IncrementRegisteredWorkflowCount()
+	return nil
 }
 
 // workflowUpdatedEvent handles the WorkflowUpdatedEvent event type.
 func (h *eventHandler) workflowUpdatedEvent(
 	_ context.Context,
-	_ WorkflowRegistryEvent,
+	_ event,
 ) error {
 	return ErrNotImplemented
 }
 
 // workflowPausedEvent handles the WorkflowPausedEvent event type.
 func (h *eventHandler) workflowPausedEvent(
 	_ context.Context,
-	_ WorkflowRegistryEvent,
+	_ event,
 ) error {
 	return ErrNotImplemented
 }
 
 // workflowActivatedEvent handles the WorkflowActivatedEvent event type.
 func (h *eventHandler) workflowActivatedEvent(
 	_ context.Context,
-	_ WorkflowRegistryEvent,
+	_ event,
 ) error {
 	return ErrNotImplemented
 }
 
 // forceUpdateSecretsEvent handles the ForceUpdateSecretsEvent event type.
 func (h *eventHandler) forceUpdateSecretsEvent(
 	ctx context.Context,
-	event WorkflowRegistryEvent,
+	event event,
 ) error {
 	// Get the URL of the secrets file from the event data
-	data, ok := event.Data.(WorkflowRegistryForceUpdateSecretsRequestedV1)
+	data, ok := event.GetData().(WorkflowRegistryForceUpdateSecretsRequestedV1)
```

Comment:
@ettec Can we move this up one level and just have the handlers accept the concrete type it needs? We're already doing a type check there; let's keep all that logic in one place.

Reply:
the problem here is the concrete type from the load and that from the event are different types,  the type from the load needs to be augmented with the event type but not the cursor and data which it does not have, seems better to interface this than set defaults which have no meaning IMO